### PR TITLE
byron proxy script generator helper

### PR DESCRIPTION
**Pull request by disassembler**
I made these scripts to make it easier to launch byron proxy with the right configs for mainnet, staging and testnet. I do not think these should permanently live here, but for now, this makes it easier to test locally for devops/devs. This is the first step to integrating into iohk-ops to deploy a continuously running proxy and chain validator.

@angerman could use a confirmation that moving nix-tools into the let block doesn't break anything, although I assume if CI passes, it's probably fine.

### CODE REVIEW

**Comment by disassembler** [comment]
spoke with @angerman and having `byronProxyScripts` in `default.nix` was causing infinite recursion because release.nix was using it for nix-tools package set. Simplest fix is to move `default.nix` -> `nix/haskell-packages.nix` and simplify `default.nix` to import that to get the byronProxy executable.

File: default.nix [hunk 1]
```diff
@@ -19,30 +19,29 @@
...
-commonLib.nix-tools.default-nix ./nix/pkgs.nix args 
-# ... and add additional non-haskell packages we want to build on CI:
-// {
-  
+
+  # Haskell Packages produced by nix-tools
+  haskellPackages = import ./nix/haskell-packages.nix {};
+  # TODO: remove when proxy and old cardano configs no
+  # longer required.
+  oldCardanoSrc = import ./nix/old-cardano.nix {
```
Comment (avieth) [inline default.nix]: Why is it called "old" cardano?

File: nix/byron-proxy-scripts.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  pkgs = commonLib.pkgs;
+  oldCardano = import oldCardanoSrc {};
+  oldCardanoLib = import (oldCardanoSrc + "/lib.nix");
+  environments = oldCardanoLib.environments;
+  loggingConfig = ../byron-proxy/cfg/logging.yaml;
+  mkTopology = relay: pkgs.writeText "topology-file" ''
+    wallet:
+      relays: [[{ host: ${relay} }]]
+      valency: 1
+      fallbacks: 7
```
Comment (avieth) [inline nix/byron-proxy-scripts.nix]: These `valency` and `fallbacks` keys actually aren't necessary: they are implicit in the dimensions of the `relays` list. Valency is the outer length, fallbacks for each entry is its length.

File: default.nix (hunk 1 above)
Comment (disassembler) [reply default.nix]: to distinguish from this repo being "new" cardano. temporary until we no longer need the "old cardano" as a proxy.

**Comment by disassembler** [comment]
DevOps team discussed, and we think systemd service configuration for nixos should live in the repo that's building it so we can run nixos tests in the future and know when a change breaks our service configuration. Eventually, this will go to cardano-shell when that's able to launch the proxy. If anyone wants to launch byron proxy on NixOS: https://github.com/disassembler/network/commit/7507d3dd330213fc76c6c3b1c365afaef927cd13 shows how it can be done.

**Comment by disassembler** [comment]
associated tests for the service: https://hydra.iohk.io/build/819073/download/1/log.html

**Pull request merged**